test/util/state_lib/simple_test.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+import sys
+import unittest
+from util.state_lib.state_machine import StateMachine
+from util.state_lib.state import State, ExitState
+import unittest
+
+
+class ForwardState(State):
+    def on_enter(self, context):
+        pass
+
+    def on_exit(self, context):
+        pass
+
+    def on_loop(self, context) -> State:
+        context.var += 1
+        context.forward_loop_count += 1
+        if context.var == 3:
+            return BackwardState()
+        return ForwardState()
+
+
+class BackwardState(State):
+    def on_enter(self, context):
+        pass
+
+    def on_exit(self, context):
+        pass
+
+    def on_loop(self, context) -> State:
+        context.var -= 1
+        context.backward_loop_count += 1
+        if context.var == 0:
+            return ExitState()
+        return BackwardState()
+
+
+from dataclasses import dataclass
+
+
+@dataclass
+class Context:
+    var: int = 0
+    forward_loop_count: int = 0
+    backward_loop_count: int = 0
+
+
+class TestSimpleStateMachine(unittest.TestCase):
+    def test_simple(self):
+        sm = StateMachine(ForwardState(), "SimpleStateMachine")
+        context = Context()
+        sm.set_context(context)
+        sm.add_transition(ForwardState(), BackwardState())
+        sm.add_transition(BackwardState(), ForwardState())
```

Comment:
why do you add this transition here if it will never actually happen? or is this just left over from testing

Reply:
yea just left over from when there was and I think I modified it so there wasn't anymore at one point